python/ml4ir/applications/ranking/tests/test_lr_schedule.py

```diff
@@ -0,0 +1,217 @@
+            decay_rate=1.0,
+        )
+        for i in range(25):
+            assert round(float(lrs(i).numpy()), 5) == 0.1
+
+    def test_cyclic_lr_in_training_pipeline(self):
+        """Test a cyclic learning rate in model training"""
+        Logger = logging_utils.setup_logging(
+            reset=True,
+            file_name=os.path.join(INPUT_DIR + 'ranklib_test_data', "output_log.csv"),
+            log_to_file=True,
+        )
+
+        io = LocalIO()
+        feature_config = self.parse_config(TFRecordTypeKey.SEQUENCE_EXAMPLE, self.feature_config_yaml_convert_to_clicks,
+                                           io)
+
+        dataset = RelevanceDataset(
+            data_dir=INPUT_DIR + '/ranklib_test_data',
+            data_format=DataFormatKey.RANKLIB,
+            feature_config=feature_config,
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            batch_size=2,
+            file_io=io,
+            preprocessing_keys_to_fns={},
+            logger=Logger,
+            keep_additional_info=KEEP_ADDITIONAL_INFO,
+            non_zero_features_only=NON_ZERO_FEATURES_ONLY,
+            max_sequence_size=319,
+        )
+
+        # Define interaction model
+        interaction_model: InteractionModel = UnivariateInteractionModel(
+            feature_config=feature_config,
+            feature_layer_keys_to_fns={},
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            max_sequence_size=319,
+            file_io=io,
+        )
+
+        # Define loss object from loss key
+        loss: RelevanceLossBase = loss_factory.get_loss(
+            loss_key=LossKey.RANK_ONE_LISTNET, scoring_type=ScoringTypeKey.POINTWISE
+        )
+
+        # Define scorer
+        scorer: ScorerBase = RelevanceScorer.from_model_config_file(
+            model_config_file=self.model_config_file,
+            interaction_model=interaction_model,
+            loss=loss,
+            logger=Logger,
+            file_io=io,
+        )
+
+        optimizer: Optimizer = get_optimizer(file_io=io, model_config_file=self.model_config_file)
+
+        # Combine the above to define a RelevanceModel
+        relevance_model: RelevanceModel = RankingModel(
+            feature_config=feature_config,
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            scorer=scorer,
+            optimizer=optimizer,
+            model_file=None,
+            file_io=io,
+            logger=Logger,
+        )
+        callbacks_list = []
+        lr_list = []
+        my_calback_object = lr_callback(lr_list)
+        callbacks_list.append(my_calback_object)
+
+        history = relevance_model.model.fit(
+            x=dataset.train,
+            validation_data=dataset.validation,
+            epochs=2,
+            verbose=True,
+            callbacks=callbacks_list,
+        )
+        lr_gold = [0.001,0.020800006,0.040599994,0.0604,0.080199994,0.1,0.080199994,0.0604,0.040599994,0.020800006,0.001,
```

Comment:
Something to keep in mind is that our tests are taking a lot of time. It is good to have tests of course. Do we need 2 epochs here? Are the data small enough? I am asking myself such questions too..

Reply:
Another question would be if we can avoid initializing the whole model stuff, and just defined the optimizer (in line 179) and be able to get its values directly without passing from fit (I get there are pros and cons everywhere).

Reply:
I set to 2 epochs to verify the behavior of scheduler for more than one epoch. The dataset I'm used is small, so time shouldn't be an issue I think.
As for using only the optimizer with no real data, I'm not sure how this would work. With that in mind, I'm not experienced with tensorflow.

Reply:
@balikasg Our tests are taking long because of save and load. I don't think training on this tiny dataset will add a lot of time. We can discuss how we want to solve the save/load tests. IMO those are the most critical tests. But I think reducing the number of parameters will solve this. Can you file a github issue so that we can work on it?

Reply:
@mohazahran I doubt you can test the optimizer without real data. We can look into it, but I like what you have here.